docs/glossary.rst

```diff
@@ -0,0 +1,90 @@
+   (`n`,)
+      A parenthesized number followed by a comma denotes a tuple with one
+      element. The trailing comma distinguishes a one-element tuple from a
+      parenthesized ``n``.
+      This is from NumPy; see https://numpy.org/doc/stable/glossary.html.
+
+   number
+      Any numeric type. eg float or int or any of the ``numpy.number``.
+
+   -like
+      Used to indicate on object of that type or that can instantiate the type.
+      E.g. :class:`~astropy.units.Quantity`-like includes ``"2 * u.km"``
+      because ``astropy.units.Quantity("2 * u.km")`` works.
+
+   unit-like
+      Must be an :class:`~astropy.units.UnitBase` (subclass) instance or a
+      string or other instance parseable by :class:`~astropy.units.Unit`.
+
+   quantity-like
+      Must be an `~astropy.units.Quantity` (or subclass) instance or a string
+      parseable by `~astropy.units.Quantity`.
+      Note that the interpretation of units in strings depends on the class --
+      ``Quantity("180d")`` is 180 **days**, while ``Angle("180d")`` is 180
+      **degrees** -- so check the string parses as intended for ``Quantity``.
+
+   angle-like
+      :term:`quantity-like`, but interpreted by an angular
+      `~astropy.units.SpecificTypeQuantity`, like `~astropy.coordinates.Angle`
+      or `~astropy.coordinates.Longitude` or `~astropy.coordinates.Latitude`.
+      Note that the interpretation of units in strings depends on the class --
+      ``Quantity("180d")`` is 180 days, while ``Angle("180d")`` is 180 degrees
+      -- so make sure the string parses as intended for ``Angle``.
+
+   length-like
+      :term:`quantity-like`, but interpretable by
+      :class:`~astropy.coordinates.Distance`.
+
+   frame-like
+      A :class:`~astropy.coordinates.BaseCoordinateFrame` subclass instance or a
+      string that can be converted to a Frame by
+      :class:`~astropy.coordinates.sky_coordinate_parsers._get_frame_class`.
+
+   coordinate-like
+      A Coordinate-type object such as a
+      :class:`~astropy.coordinates.BaseCoordinateFrame` subclass instance or a
+      :class:`~astropy.coordinates.SkyCoord` (or subclass) instance.
+
+   table-like
+      An astropy :class:`~astropy.table.Table` or any object that can
+      initialize one. Anything marked as table-like will be processed through
+      a :class:`~astropy.table.Table`.
+
+   time-like
+      :class:`~astropy.time.Time` or any valid initializer.
+
+   buffer-like
+      Anything that implements Python's buffer protocol. See
+      https://docs.python.org/3/c-api/buffer.html#bufferobjects
+
+   writable file-like object
+      In the context of a :term:`python:file-like object` object, anything
+      that supports writing with a method ``write``.
+
+   readable file-like object
+      In the context of a :term:`python:file-like object` object, anything
+      that supports writing with a method ``read``.
+
+   Base HDU
+      The fundamental HDU data structure.
+      Defined in ``astropy.io.fits.hdu``.
```

Comment:
```suggestion
   HDU
      Header Data Unit, the fundamental container structure of the FITS format
      consisting of a ``data`` member and its associated metadata in a ``header``.
      Defined in ``astropy.io.fits.hdu``.
```
I am putting this up for discussion as a suggestion to put a functional explanation here. It is unfortunate that Xref cannot insert a link to the actual documentation page, but nonetheless I assume just explaining what a `HDU` is (which pretty much every user of the format will have or should have heard of) is more useful. Having it defined in the `_BaseHDU` meta-class is a purely internal implementation detail, which most users will never be aware of, since the class is private.
@saimn what do you think?

Reply:
Alternatively...  I've been looking at the numpydoc backend and it appears that we can instead xref to a `:ref:` in the docs. I don't see a good place in the docs (maybe somewhere in https://docs.astropy.org/en/stable/io/fits/index.html), but were

```
Header Data Unit, the fundamental container structure of the FITS format
consisting of a ``data`` member and its associated metadata in a ``header``.
Defined in ``astropy.io.fits.hdu``.
```

in the docs with an RST link, we could skip on adding to the glossary.

Reply:
Yes, I guess we could have a small subsection at the beginning of "Getting started" in https://docs.astropy.org/en/stable/io/fits/index.html explaining what a HDU is. There is currently a sentence which is similar to the enhanced version from @dhomeier:
> An HDU (Header Data Unit) is the highest level component of the FITS file structure, consisting of a header and (typically) a data array or table.

Reply:
The best summary (and the main documentation section on it) imo is still [hdus.rst](https://docs.astropy.org/en/stable/io/fits/api/hdus.html); if linking this as a `:ref:` works now, great!

Reply:
WIP. almost ready to push 🤞 .

Reply:
got it working. My one reservation is that astropy has deleted itself from the inter-sphinx mapping. A potential followup would be to link to the HDU page in a project-agnostic manner by going through an inter-sphinx mapping. This is more friendly to downstream projects as they won't have to remap the HDU link in their own conf.py.

Reply:
Added as comment in #11555

Reply:
<img width="862" alt="Screen Shot 2021-04-16 at 00 30 11" src="https://user-images.githubusercontent.com/8949649/115094567-00edff80-9eec-11eb-8e4d-5f59e43554e3.png">

It works!

Reply:
@dhomeier, look good?

Reply:
The font is a little different between the numpydoc xref links and the direct ReST links. I can fix this in a followup. Fiddling with a CSS file.

Reply:
Great, thanks! I could not even spot any font differences in Safari.
The plural in the original caption of hdus did not make much sense to me, so I took the opportunity to correct that as well.

Reply:
@dhomeier. I implemented the suggestions below, thanks!